src/core/pointcloud/qgspointcloudindex.h

```diff
@@ -174,6 +174,13 @@ class CORE_EXPORT QgsPointCloudIndex: public QObject
     explicit QgsPointCloudIndex();
     ~QgsPointCloudIndex();
 
+    /**
+     * Returns a clone of the current point cloud index object
+     * \note It is the responsibility of the caller to handle the ownership and delete the object.
+     * \since QGIS 3.26
+     */
+    virtual QgsPointCloudIndex *clone() const = 0;
```

Comment:
Since this isn't exposed to python, would be better to make it return a unique_ptr value so that ownership.by the caller is forced